external-crates/move/crates/move-package/src/lock_file/schema.rs

```diff
@@ -136,3 +150,46 @@ pub(crate) fn write_prologue(
 
     Ok(())
 }
+
+pub fn update_compiler_toolchain(
+    file: &mut LockFile,
+    compiler_version: String,
+    build_config: &BuildConfig,
+) -> Result<()> {
+    let mut toml_string = String::new();
+    file.read_to_string(&mut toml_string)?;
+    file.seek(SeekFrom::Start(0))?;
+    let mut toml = toml_string.parse::<toml_edit::Document>()?;
+    let move_table = toml["move"].as_table_mut().ok_or(std::fmt::Error)?;
+    move_table["compiler-version"] = toml_edit::value(compiler_version);
+    let compiler_flags = toml::to_string(build_config)?.parse()?;
+    let compiler_flags = to_toml_edit_value(&compiler_flags);
+    move_table["compiler-flags"] = toml_edit::value(compiler_flags);
+    write!(file, "{}", toml)?;
+    file.flush()?;
+    Ok(())
+}
+
+fn to_toml_edit_value(value: &toml::Value) -> toml_edit::Value {
```

Comment:
The existence of all these `Value::from` implementations in `toml_edit` makes me wonder if you can't just do:

```rust
let value: toml_edit::Value = value.clone().into();
```

Reply:
Alas, looks like no. The trouble (depending how I try to make use of `Value::from`):

- If trying to convert directly from `toml::Value` => `toml_edit::Value`, there isn't a converter. This possibly / probably because `toml_edit` doesn't implement a converter because it defines its `toml_edit::Value` types in a different way (there is no `Table` for this type, only `InlineTable` -- there is another `toml_edit::Item` type defines `Table`, and wraps `toml_edit::Value`. And unfortunatley, neither does a `toml_edit::Item::from` or any other `toml_edit` enum implement the work we want it to do). 

- If converting from a string toml::Value, toml_edit just outputs a multiline string of the struct, not the toml types we want.

Digging into this though, the good news is I arrived at (a) we don't need to convert-to-string and then parse, I can just use `try_into` on the Rust type and (b) we can emit `toml_edit::Item` values, which permits the non-inline table (which I also prefer!).